### PR TITLE
fix: Better handle octet stream case

### PR DESCRIPTION
**Pull request by igoroctaviano**
### Context

- https://github.com/cornerstonejs/cornerstone3D/issues/2605

When the data source is configured with `acceptHeader: ['application/octet-stream']` (single-part), stack viewports for CR studies (and other raw DICOM) would show a black screen. OHIF always used `streaming: true` for stack loading, which is designed for HTJ2K progressive decode. With octet-stream, the server returns raw DICOM and the streaming path delivers partial data to the decoder. Uncompressed CR cannot be decoded from partial data, so decoding failed and nothing rendered.

#### extensions/cornerstone/src/index.tsx
- Added `extensionManager` to the `onModeEnter` callback arguments.
- Read the active data source config and its `acceptHeader`.
- If `acceptHeader` is only `['application/octet-stream']`, use `streaming: false` for stack retrieve options (full-file XHR load).
- Otherwise keep `streaming: true` (existing `stackRetrieveOptions`).

**Results:**
- CR studies and other raw DICOM load and display correctly with octet-stream.
- HTJ2K studies still use streaming and progressive loading when the data source uses multipart.

### Testing

1. Use a data source with `acceptHeader: ['application/octet-stream']`.
2. Open a CR study and verify images display.
3. Switch to a data source with multipart/related (or without octet-stream) and verify HTJ2K streaming still works.
4. Compare loading behavior and image quality for both configurations.

#### PR

- [] My Pull Request title is descriptive, accurate and follows the
  semantic-release format and guidelines.
  

#### Code

- [] My code has been well-documented (function documentation, inline comments,
  etc.)

#### Public Documentation Updates

- [] The documentation page has been updated as necessary for any public API
  additions or removals.

#### Tested Environment

- [] OS: 
- [] Node version: 
- [] Browser:
  

[blog]: https://circleci.com/blog/triggering-trusted-ci-jobs-on-untrusted-forks/
[script]: https://github.com/jklukas/git-push-fork-to-upstream-branch

<h2>Greptile Overview</h2>

<h3>Greptile Summary</h3>

Fixes black screen issue for CR studies when using `application/octet-stream` accept header by conditionally disabling streaming for stack loading.

- Added `extensionManager` parameter to `onModeEnter` callback to access data source configuration
- Detects when data source uses only `application/octet-stream` accept header
- Disables streaming (`streaming: false`) for octet-stream to enable full-file loading before decode
- Maintains streaming for HTJ2K progressive loading with multipart/related responses
- Uses defensive optional chaining to safely access data source config

<h3>Confidence Score: 4/5</h3>

- This PR is safe to merge with minor considerations around edge cases
- The fix addresses a specific issue with proper logic and defensive coding. The use of optional chaining prevents runtime errors. The change is isolated to the stack loading configuration and doesn't affect other functionality. Score reduced from 5 due to lack of explicit tests for this edge case and potential for inconsistent behavior if data source config changes during runtime
- No files require special attention

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| extensions/cornerstone/src/index.tsx | Added logic to conditionally disable streaming for octet-stream data sources; added `extensionManager` parameter to `onModeEnter` |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant Mode as Mode
    participant Extension as Cornerstone Extension
    participant ExtMgr as Extension Manager
    participant DataSource as Data Source
    participant Provider as Image Retrieve Metadata Provider
    
    Mode->>Extension: onModeEnter({extensionManager, ...})
    Extension->>ExtMgr: getActiveDataSource()
    ExtMgr-->>Extension: [dataSource]
    Extension->>DataSource: getConfig()
    DataSource-->>Extension: {acceptHeader: [...]}
    
    alt acceptHeader is ['application/octet-stream']
        Extension->>Extension: Set stackOptions with streaming: false
        Note over Extension: Full-file XHR load for raw DICOM
    else acceptHeader has other values or multipart
        Extension->>Extension: Use stackRetrieveOptions with streaming: true
        Note over Extension: Progressive HTJ2K decode
    end
    
    Extension->>Provider: add('stack', stackOptions)
    Note over Provider: Stack loading configured based on accept header
```

<sub>(3/5) Reply to the agent's comments like "Can you suggest a fix for this @greptileai?" or ask follow-up questions!</sub>

### CODE REVIEW

File: extensions/cornerstone/src/index.tsx [hunk 1]
```diff
@@ -131,10 +135,22 @@ const cornerstoneExtension: Types.Extensions.Extension = {
...
+     * (single-part). Streaming only benefits HTJ2K progressive decode; octet-stream
+     * returns raw DICOM (e.g. CR) which requires full file before decode.
+     */
+    const sourceConfig = extensionManager?.getActiveDataSource?.()?.[0]?.getConfig?.() ?? {};
+    const acceptHeader = sourceConfig.acceptHeader ?? [];
+    const isOctetStreamOnly =
+      Array.isArray(acceptHeader) &&
+      acceptHeader.length > 0 &&
+      acceptHeader.every((h) => h === 'application/octet-stream');
+    const stackOptions = isOctetStreamOnly
```
Comment (wayfarer3130) [inline extensions/cornerstone/src/index.tsx]: This isn't a good general fix because it doesnt deal with non-streaming returns just being the selected response type.  
I would accept a configuration option to set the full default retrieve options in the configuration, which could then be used to disable streaming entirely - it should be done as a customization for the default streaming value.
The better long term fix is to prevent the black image display on load for uncompressed images when streaming is enabled.  The problem is that you cant tell if streaming is allowed or not on a transfer syntax until you have started retrieving because you don't know the transfer syntax for sure until then.  I thought that case had been handled automatically in CS3D.  I'm looking at that now but I don't see hte handling for it.

Comment (igoroctaviano) [reply extensions/cornerstone/src/index.tsx]: Makes sense, I just made a change to read from stack options.

File: extensions/cornerstone/src/index.tsx [hunk 2]
```diff
@@ -137,19 +138,17 @@ const cornerstoneExtension: Types.Extensions.Extension = {
...
     const sourceConfig = extensionManager?.getActiveDataSource?.()?.[0]?.getConfig?.() ?? {};
-    const acceptHeader = sourceConfig.acceptHeader ?? [];
-    const isOctetStreamOnly =
-      Array.isArray(acceptHeader) &&
-      acceptHeader.length > 0 &&
-      acceptHeader.every((h) => h === 'application/octet-stream');
-    const stackOptions = isOctetStreamOnly
-      ? { retrieveOptions: { single: { streaming: false, decodeLevel: 1 } } }
-      : stackRetrieveOptions;
+    const config = sourceConfig.stackRetrieveOptions;
```
Comment (wayfarer3130) [inline extensions/cornerstone/src/index.tsx]: Just get the customized value from the customization service directly, and you don't need to worry about any merge options - the customization framework already takes care of that.

Comment (wayfarer3130) [reply extensions/cornerstone/src/index.tsx]: How about immutability helper since we are moving towards that for customization:
const stackOptions = update(sourceConfig.stackRetrieveOptions, DEFAULT_STACK_RETRIEVE_OPTIONS)

Comment (igoroctaviano) [reply extensions/cornerstone/src/index.tsx]: Done!

File: extensions/cornerstone/src/index.tsx [hunk 3]
```diff
@@ -131,10 +136,16 @@ const cornerstoneExtension: Types.Extensions.Extension = {
...
-    imageRetrieveMetadataProvider.add('stack', stackRetrieveOptions);
+
+    /**
+     * Stack retrieve options: read from active data source configuration.
+     * Merged with defaults. Set streaming: false in data source config for uncompressed
+     * DICOM that requires full file before decode, to avoid black image on load.
+     */
+    const sourceConfig = extensionManager?.getActiveDataSource?.()?.[0]?.getConfig?.() ?? {};
+    const config = sourceConfig.stackRetrieveOptions ?? {};
+    const stackOptions = update(DEFAULT_STACK_RETRIEVE_OPTIONS, { $merge: config }) as typeof DEFAULT_STACK_RETRIEVE_OPTIONS;
```
Comment (wayfarer3130) [inline extensions/cornerstone/src/index.tsx]: Can you leave the $merge for inside stackRetrieveOptions?  That way someone can do $set or any of the operations to replace it from the top?
